### DISCUSSION TITLE
Can a plugin store information in the database?

### DISCUSSION THREAD

**Discussion by CyberSmash** [NationalSecurityAgency/ghidra#3855] [Q&A]
I have a need for a plugin to save user-supplied data in some form that is associated with a project. For the sake of discussion you can imagine that this data is a notebook for sharing information between teams of people working on the same program. (I'm aware the comment system could fulfill this same role, but suffice it to say that I want to develop a plugin that saves some amount of user supplied data).

My first idea was to do whatever the bookmarks system was doing, however on investigation it looks as if many of the core features of the code viewer (listing) are hardwired into in the `managers` array of the `ProgramDB` class of `ProgramDB.java`. All managers seem to need to be initialized using this interface, but I don't see any way of adding another database manager from my plugin at run time.

Is there a way to extend the database for my use that I'm not seeing or is there a better way of saving user-supplied information without using the database that can also be passed around from person to person who uses the project?

**Comment by astrelsky** [comment]
https://github.com/astrelsky/Ghidra-Cpp-Class-Analyzer/blob/master/src/main/java/cppclassanalyzer/data/manager/ClassTypeInfoManagerDB.java#L64

AFAIK the only implementation of `Program` is `ProgramDB`.

There is also `Program.getUsrPropertyManager` for which you could use an `ObjectPropertyMap`. If it is small and having the objects class path saved to the db isn't a problem then you should go this route.
